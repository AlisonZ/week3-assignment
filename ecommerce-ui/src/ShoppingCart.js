import React, {Component} from 'react';
import CartItem from './CartItem';
import cartImg from './shopping-cart.png';
import PropTypes from 'prop-types';
import removeIcon from './remove-icon.png';


class ShoppingCart extends Component {
    static propTypes = {
        selectedRentals: PropTypes.array,
        removeFromCart: PropTypes.func,
    }

    constructor() {
        super()
        this.state = {
            isOpen: false,
        }

        this.handleDropDown = this.handleDropDown.bind(this);   
    }

    handleDropDown = () => {
        this.setState({
            isOpen: !this.state.isOpen,
        });
    }

    render() {
        const rentalItems = this.props.selectedRentals.map((rental, idx) => 
        <li key={idx}>
            <CartItem 
                rental={rental} 
            />
            <img src={removeIcon} onClick={() => this.props.removeFromCart(idx)} className="cart-remove-img cart-details" alt="minus-symbol"/>

        </li>
        );

        let price = 0;
        this.props.selectedRentals.map((rental) =>
        price += rental.payment.cost
        );

        return (    
            <span >
                <img src={cartImg} onClick={this.handleDropDown} className="cart-img" alt="shopping-cart"/>
                <div>{
                    this.state.isOpen ?
                    (<div>
                        <ul className="unstyled-list">
                            {rentalItems}
                        </ul>
                        <div className="light-grey-primary-title">Total= ${price}</div>
                    </div>) : null
                }
                </div>
            </span>
        );
    }
}

export default ShoppingCart;