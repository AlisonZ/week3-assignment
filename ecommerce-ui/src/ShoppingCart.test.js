import React from 'react';
import {shallow} from 'enzyme';
import ShoppingCart from './ShoppingCart';
import {MOCK_RENTAL_1, MOCK_RENTAL_2} from './TestMocks';


it('shallow renders a Shopping Cart', () => {
    const wrapper = shallow(
    <ShoppingCart 
        selectedRentals={[MOCK_RENTAL_1, MOCK_RENTAL_2]}
    />);

    expect(wrapper).toMatchSnapshot();
});