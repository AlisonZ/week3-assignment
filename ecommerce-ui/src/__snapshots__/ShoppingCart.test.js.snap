// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders a Shopping Cart 1`] = `
<span>
  <img
    alt="shopping-cart"
    className="cart-img"
    onClick={[Function]}
    src="shopping-cart.png"
  />
  <div />
</span>
`;
